Artesian/Artesian.SDK.Tests/SerializationMigrationTests.cs

```diff
@@ -0,0 +1,640 @@
+// Copyright (c) ARK LTD. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for
+// license information.
+#pragma warning disable MA0002 // Dictionary ContainsKey is fine without comparer in tests
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+using Artesian.SDK.Common;
+using Artesian.SDK.Dto;
+using Artesian.SDK.Dto.UoM;
+using Artesian.SDK.Service;
+using NodaTime;
+using NodaTime.Serialization.JsonNet;
+using NodaTime.Serialization.SystemTextJson;
+using Newtonsoft.Json;
+using NUnit.Framework;
+
+namespace Artesian.SDK.Tests
+{
+    /// <summary>
+    /// Serialization tests to ensure compatibility during migration from Newtonsoft.Json to System.Text.Json
+    /// These tests validate that System.Text.Json can:
+    /// 1. Deserialize JSON created by Newtonsoft.Json (backward compatibility)
+    /// 2. Serialize objects to JSON that is equivalent to Newtonsoft.Json output (forward compatibility)
+    /// </summary>
+    [TestFixture]
+    public class SerializationMigrationTests
+    {
+        private JsonSerializerSettings _newtonsoftSettings = null!;
+        private JsonSerializerOptions _stjOptions = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            // Configure Newtonsoft.Json settings matching the original Client.cs setup
+            _newtonsoftSettings = new JsonSerializerSettings();
```

Comment:
check again that newtonsoft settings are identical to the one that are currently on master in the Serializer. this is very important as otherwise we're testing correctly that we're testing regression correctly.